### PR TITLE
Fix for #34422: bump C* version to 3.9

### PR DESCRIPTION
**Pull request by mstump**
Fix for #34422

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37145)

### CODE REVIEW

**Comment by dims** [comment]
@k8s-bot ok to test

**Comment by mstump** [comment]
It seems from the logs that the Jenkins automation failed in a way that's unrelated to the PR. What are the next steps?

**Comment by foxish** [comment]
> W1119 13:16:25.878] FAIL: changes needed but not made due to --verify
> I1119 13:16:25.880] /go/src/k8s.io/kubernetes/examples/ is out of date. Please run hack/update-munge-docs.sh
> I1119 13:16:25.882] FAILED   hack/make-rules/../../hack/verify-munge-docs.sh	16s

@mstump, looks like the failure is related this PR

**Comment by foxish** [comment]
cc @kubernetes/sig-apps

File: examples/storage/cassandra/image/Dockerfile [hunk 1]
```diff
@@ -20,28 +20,31 @@ ENV DI_VERSION="1.1.1" DI_SHA="dec8167091671df0dd3748a8938102479db5fffc"
...
   && mkdir -p /cassandra_data/data \
-  && mv /logback.xml /cassandra.yaml /etc/cassandra/ \
+  && mv /logback.xml /cassandra.yaml /jvm.options /etc/cassandra/
 
   # Not able to run as cassandra until https://github.com/kubernetes/kubernetes/issues/2630 is resolved
   # && chown -R cassandra: /etc/cassandra /cassandra_data /run.sh /kubernetes-cassandra.jar \
   # && chmod o+w -R /etc/cassandra /cassandra_data \
 
-  && apt-get -y purge curl localepurge \
+RUN apt-get -y purge curl localepurge \
```
Comment (chrislovecnm) [inline examples/storage/cassandra/image/Dockerfile]: Please only use one RUN ;) any reason you have two. Less layers ;)

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
Maven file updates for the new version? Looks great otherwise.

**Comment by chrislovecnm** [comment]
@foxish can u push the docker? Otherwise the e2e tests are borked.

File: examples/storage/cassandra/image/Dockerfile [hunk 2]
```diff
@@ -20,28 +20,31 @@ ENV DI_VERSION="1.1.1" DI_SHA="dec8167091671df0dd3748a8938102479db5fffc"
...
   && apt-get update \
-  && apt-get -qq -y install --no-install-recommends procps openjdk-8-jre-headless libjemalloc1 curl \
-    localepurge \
-  && curl -L https://github.com/Yelp/dumb-init/releases/download/v${DI_VERSION}/dumb-init_${DI_VERSION}_amd64 > /sbin/dumb-init \
-  && echo "$DI_SHA  /sbin/dumb-init" | sha1sum -c - \
-  && mv /cassandra.list /etc/apt/sources.list.d/cassandra.list \
+  && apt-get -qq -y --force-yes install --no-install-recommends procps openjdk-8-jre-headless libjemalloc1 curl localepurge
+
+RUN curl -L https://github.com/Yelp/dumb-init/releases/download/v${DI_VERSION}/dumb-init_${DI_VERSION}_amd64 > /sbin/dumb-init \
+  && echo "$DI_SHA  /sbin/dumb-init" | sha1sum -c -
```
Comment (chrislovecnm) [inline examples/storage/cassandra/image/Dockerfile]: And just one Run. Every run is another layer :P

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
Please let's do one RUN

**Comment by mstump** [comment]
I incorporated the feedback about the RUN statement in the docker file. I also took out the version bump in the statefulset example, I believe that was that cause of the original build error. Someone with privileges will need to bump that version and push to docker as a followup action.

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
Still missing the maven Pom file updates. We need to bump the dependency version for the jar as well.

**Comment by chrislovecnm** [comment]
We need to maven pom updated and the jar built before we push the docker version.

**Comment by kow3ns** [comment]
Reviewed 4 of 6 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[examples/storage/cassandra/image/files/kubernetes-cassandra.jar, line 0 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/37145#-KX7gh8I1-tmEcb6_man:-KX7gh8I1-tmEcb6_mao:bx6h5ha) ([raw file](https://github.com/kubernetes/kubernetes/blob/d80c1df221b14d9fdad447a5e7478560f86797f6/examples/storage/cassandra/image/files/kubernetes-cassandra.jar#L0)):*
I believe chris already commented on this but Kubernetes SeedProvider needs to be rebuilt against cassandra-all 3.9. The cassandra-version property needs to be set in the projects pom.  The SeedProvider needs to be rebuilt and copied into the Docker image.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/37145)*

**Comment by mstump** [comment]
Ok, hopefully this is the last update. Bumped the version in the POM and rebuilt the jar.

**Comment by chrislovecnm** [comment]
@janetkuo can you deploy the new container for this PR?

@mstump LGTM, let's get the container changed, and then you can put another PR in  for the manifests.

**Comment by kow3ns** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/37145#-:-KXhEhQjwulOhspTFCeo:bnfp4nl)*

**Comment by chrislovecnm** [comment]
@kow3ns who can upload the container to gcr.io?

**Comment by kow3ns** [comment]
I should be able to

---







*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/37145#-:-KXvzT_xIvsHyK_2Zvi1:b2pbotx)*

**Comment by kow3ns** [comment]
/lgtm

root@cassandra-0:/# nodetool version
ReleaseVersion: 3.9

root@cassandra-0:/# nodetool status
Datacenter: DC1-K8Demo
======================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address     Load       Tokens       Owns (effective)  Host ID                               Rack
UN  10.244.3.4  83.17 KiB  32           71.2%             6d79d831-16c6-49ff-8b92-22d6dd20e48c  Rack1-K8Demo
UN  10.244.1.4  65.61 KiB  32           65.0%             b269337d-6503-43f5-b67d-d697e4926839  Rack1-K8Demo
UN  10.244.2.9  103.84 KiB  32           63.8%             d442393d-99db-4fc9-85ff-d43c73f4bae7  Rack1-K8Demo

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/37145#-:-KY10Z2LZpHvKz_z4KwX:b-e5xwyj)*

**Comment by kow3ns** [comment]
Reviewed 1 of 6 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/37145)*

**Comment by chrislovecnm** [comment]
/lgtm

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/37145#-:-KY5kxepXAXOVjSOuWBz:b-8ks5qa)*

**Comment by rmmh** [comment]
@k8s-bot unit test this

**Pull request merged**